build.cake

```diff
@@ -0,0 +1,408 @@
+    }
+});
+
+Task("TestBuild")
+    .IsDependentOn("Restore")
+    .Does(() =>
+{
+    foreach (var pair in buildPlan.TestProjects)
+    {
+        foreach (var framework in pair.Value)
+        {
+            var project = pair.Key;
+            var process = StartAndReturnProcess(dotnetcli, 
+                new ProcessSettings
+                { 
+                    Arguments = $"build --framework {framework} --configuration {testConfiguration} {testFolder}/{project}",
+                    RedirectStandardOutput = true
+                });
+            process.WaitForExit();
+            FileWriteLines($"{logFolder}/{project}-{framework}-build.log", process.GetStandardOutput().ToArray());
+        }
+    }
+});
+
+Task("TestCore")
+    .IsDependentOn("TestBuild")
+    .Does(() =>
+{
+    foreach (var pair in buildPlan.TestProjects)
+    {
+        foreach (var framework in pair.Value)
+        {
+            if (!framework.Equals("dnxcore50"))
+            {
+                continue;
+            }
+            
+            var project = pair.Key;
+            var exitCode = StartProcess(dotnetcli, 
+                new ProcessSettings
+                { 
+                    Arguments = $"test -xml {logFolder}/{project}-{framework}-result.xml -notrait category=failing",
+                    WorkingDirectory = $"{testFolder}/{project}"
+                });
+            if (exitCode != 0)
+            {
+                throw new Exception($"Test failed {project} / {framework}");
+            }
+        }
+    }
+});
+
+Task("Test")
+    .IsDependentOn("TestBuild")
+    .Does(() =>
+{
+    foreach (var pair in buildPlan.TestProjects)
+    {
+        foreach (var framework in pair.Value)
+        {
+            if (framework.Equals("dnxcore50"))
+            {
+                continue;
+            }
+
+            var project = pair.Key;
+            var runtime = GetRuntimeInPath($"{testFolder}/{project}/bin/{testConfiguration}/{framework}/*");
+            var instanceFolder = $"{testFolder}/{project}/bin/{testConfiguration}/{framework}/{runtime}";
+            CopyFileToDirectory($"{toolsFolder}/xunit.runner.console/tools/xunit.console.exe", instanceFolder);
+            CopyFileToDirectory($"{toolsFolder}/xunit.runner.console/tools/xunit.runner.utility.desktop.dll", instanceFolder);
+            var logFile = $"{logFolder}/{project}-{framework}-result.xml";
+            var xunitSettings = new XUnit2Settings
+            {
+                ToolPath = $"{instanceFolder}/xunit.console.exe",
+                ArgumentCustomization = builder =>
+                {
+                    builder.Append("-xml");
+                    builder.Append(logFile);
+                    return builder;
+                }
+            };
+            xunitSettings.ExcludeTrait("category", new[] { "failing" });
+            XUnit2($"{instanceFolder}/{project}.dll", xunitSettings);
+        }
+    }
+});
+
+Task("OnlyPublish")
```

Comment:
The system does it implicitly by running the RestrictToLocalRuntime task at the start of the plan, but it makes sense to make it specifically for Publish. It is only Publish that is affected by this anyway.